aten/src/ATen/native/cpu/AvgPoolKernel.cpp

```diff
@@ -531,13 +531,596 @@ void avg_pool2d_backward_kernel_impl(
   switch (grad_output.suggest_memory_format()) {
     case at::MemoryFormat::Contiguous: {
       AT_DISPATCH_FLOATING_TYPES_AND3(kLong, kBFloat16, kHalf, grad_output.scalar_type(), "avg_pool2d_backward", [&] {
-        cpu_avg_pool_backward<scalar_t>(grad_input, grad_output, kW, kH, dW, dH, padW, padH, count_include_pad, divisor_override);
+        cpu_avg_pool2d_backward<scalar_t>(grad_input, grad_output, kW, kH, dW, dH, padW, padH, count_include_pad, divisor_override);
       });
       break;
     }
     case at::MemoryFormat::ChannelsLast: {
       AT_DISPATCH_FLOATING_TYPES_AND3(kLong, kBFloat16, kHalf, grad_output.scalar_type(), "avg_pool2d_backward_channels_last", [&] {
-        cpu_avg_pool_backward_channels_last<scalar_t>(grad_input, grad_output, kW, kH, dW, dH, padW, padH, count_include_pad, divisor_override);
+        cpu_avg_pool2d_backward_channels_last<scalar_t>(grad_input, grad_output, kW, kH, dW, dH, padW, padH, count_include_pad, divisor_override);
+      });
+      break;
+    }
+    default:
+      TORCH_CHECK(false, "Unsupported memory format. Supports only ChannelsLast, Contiguous");
+  }
+}
+
+
+template <typename scalar_t>
+void cpu_avg_pool3d(
```

Comment:
What do you think a good code review strategy for these kernels are? Are they similar to their non-channels last brethren and I just need to diff them, or is there more to it?

Reply:
This is purely copied and adapted code of `cpu_avg_pool`, `cpu_avg_pool_channels_last`, which are the 2d counterparts. I wrote this by copying those functions then writing the 3d part of it. Thus I believe you can diff those with the previous version of those function, or even diff those with the renamed 2d version.

Reply:
Thank you for giving this your time !